src/FieldWarning/Assets/Model/Armory/Armory.cs

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2017-present, PFW Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is
+ * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See
+ * the License for the specific language governing permissions and limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using UnityEngine;
+
+namespace PFW.Model.Armory
+{
+    /// <summary>
+    ///     The parsed and dereferenced model containing all of our
+    ///     unit data.
+    /// </summary>
+    public class Armory
+    {
+        public readonly List<Unit>[] Categories;
+        public readonly Dictionary<string, Unit> Units;
+
+        public Armory(ArmoryConfig armoryConfig)
+        {
+            Categories = new List<Unit>[(int)UnitCategory._SIZE];
```

Comment:
So I was thinking it makes sense for category to be an enum but then I thought about making it more dynamic. We should read all the possible categories directory from the armory json files. We shouldnt constrain ourselves with Enums.

Reply:
I think it makes the code harder to read and in the end there's stuff like the deckbuilder UI which won't be done dynamically (we could do it dynamically, but the gains don't seem worth it?). For now I'd keep it static just for readability and because dynamic categories isn't that useful as a feature.